util/app/discovery/services.go

```diff
@@ -0,0 +1,121 @@
+package discovery
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"slices"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func (s *plugins) namespace() string {
+	ns := os.Getenv("ARGOCD_SERVICE_PLUGINS_NAMESPACE")
+
+	switch ns {
+	case `*`:
+		return corev1.NamespaceAll
+	case "":
+		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+		configOverrides := &clientcmd.ConfigOverrides{}
+		kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+		namespace, _, err := kubeConfig.Namespace()
+		if err != nil {
+			log.Warnf("Error getting default namespace %s. Got %s", err, namespace)
+		}
+		return namespace
+	default:
+		return ns
+	}
+}
+
+func (s *plugins) serviceWatcher(c *kubernetes.Clientset) {
+	//	labelOptions := informers.WithTweakListOptions()
+	factory := informers.NewFilteredSharedInformerFactory(c, 10*time.Minute, s.namespace(),
+		func(opts *metav1.ListOptions) {
+			opts.LabelSelector = "argocd.argoproj.io/plugin=true"
+		})
+
+	informer := factory.Core().V1().Services()
+	s.informer = &informer
+
+	_, err := informer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    s.svcAdd,
+			UpdateFunc: s.svcUpdate,
+			DeleteFunc: s.svcDelete,
+		},
+	)
+	if err != nil {
+		log.Errorf("Failed to initialise plugin services watcher, plugins as services will not work: %s", err)
```

Comment:
```suggestion
		log.Errorf("Failed to initialize plugin services watcher, plugins as services will not work: %s", err)
```
(this is the spelling currently used in this repo)